examples/linear_model/plot_tweedie_regression_insurance_claims.py

```diff
@@ -0,0 +1,593 @@
+            else:
+                y_pred = estimator.predict(X)
+
+            if metric is None:
+                if not hasattr(estimator, "score"):
+                    continue
+                score = estimator.score(X, y, _weights)
+            else:
+                score = metric(y, y_pred, _weights)
+
+            res.append(
+                {"subset": subset_label, "metric": score_label, "score": score}
+            )
+
+    res = (
+        pd.DataFrame(res)
+        .set_index(["metric", "subset"])
+        .score.unstack(-1)
+        .round(2)
+        .loc[:, ['train', 'test']]
+    )
+    return res
+
+
+##############################################################################
+#
+# Loading datasets, basic feature extraction and target definitions
+# -----------------------------------------------------------------
+#
+# We construct the freMTPL2 dataset by joining the freMTPL2freq table,
+# containing the number of claims (``ClaimNb``), with the freMTPL2sev table,
+# containing the claim amount (``ClaimAmount``) for the same policy ids
+# (``IDpol``).
+
+df = load_mtpl2(n_samples=60000)
+
+# Note: filter out claims with zero amount, as the severity model
+# requires strictly positive target values.
+df.loc[(df["ClaimAmount"] == 0) & (df["ClaimNb"] >= 1), "ClaimNb"] = 0
+
+# Correct for unreasonable observations (that might be data error)
+# and a few exceptionally large claim amounts
+df["ClaimNb"] = df["ClaimNb"].clip(upper=4)
+df["Exposure"] = df["Exposure"].clip(upper=1)
+df["ClaimAmount"] = df["ClaimAmount"].clip(upper=200000)
+
+log_scale_transformer = make_pipeline(
+    FunctionTransformer(np.log, validate=False),
+    StandardScaler()
+)
+
+column_trans = ColumnTransformer(
+    [
+        ("binned_numeric", KBinsDiscretizer(n_bins=10),
+            ["VehAge", "DrivAge"]),
+        ("onehot_categorical", OneHotEncoder(),
+            ["VehBrand", "VehPower", "VehGas", "Region", "Area"]),
+        ("passthrough_numeric", "passthrough",
+            ["BonusMalus"]),
+        ("log_scaled_numeric", log_scale_transformer,
+            ["Density"]),
+    ],
+    remainder="drop",
+)
+X = column_trans.fit_transform(df)
+
+# Insurances companies are interested in modeling the Pure Premium, that is
+# the expected total claim amount per unit of exposure for each policyholder
+# in their portfolio:
+df["PurePremium"] = df["ClaimAmount"] / df["Exposure"]
+
+# This can be indirectly approximated by a 2-step modeling: the product of the
+# Frequency times the average claim amount per claim:
+df["Frequency"] = df["ClaimNb"] / df["Exposure"]
+df["AvgClaimAmount"] = df["ClaimAmount"] / np.fmax(df["ClaimNb"], 1)
+
+print(df[df.ClaimAmount > 0].head())
+
+##############################################################################
+#
+# Frequency model -- Poisson distribution
+# ---------------------------------------
+#
+# The number of claims (``ClaimNb``) is a positive integer that can be modeled
```

Comment:
`positive integer that can be modelled as a Poisson distribution` -> IMO, the sentence seems weird since we model a scalar to a distribution.

```
The number of claims (``ClaimNB``) is a positive integer. Thus, this target 
can be modelled by a Poisson distribution.
```